server/src/core/tenant/datasource/entities/data-source-metadata.entity.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  Column,
+  CreateDateColumn,
+  Entity,
+  PrimaryGeneratedColumn,
+  UpdateDateColumn,
+} from 'typeorm';
+
+// export const dataSourceEntity = new EntitySchema({
```

Comment:
This is basically how it looks like for custom objects when they are dynamically generated. For metadata entities we can use the decorator way below because it looks nicer but the commented part works as well.